router/core/router.go

```diff
@@ -222,6 +222,15 @@ func NewRouter(opts ...Option) (*Router, error) {
 		r.playgroundConfig.Path = "/"
 	}
 
+	// handle introspection config deprecation
+	// if either the old deprecated or the new config is set to false, introspection is disabled
+	if !r.introspection {
+		r.introspectionConfig.Enabled = r.introspection
+		r.logger.Warn("The introspection_enabled option is deprecated. Use the introspection.enabled option in the config instead.")
+	} else if !r.introspectionConfig.Enabled {
+		r.introspection = r.introspectionConfig.Enabled
+	}
```

Comment:
Nit: I think elsewhere we use the JSON schema path format like `/introspection/enabled`? If so we should be consistent.